core/src/main/java/org/elasticsearch/index/mapper/geo/GeoPointFieldMapper.java

```diff
@@ -193,133 +99,36 @@ public Builder fieldDataSettings(Settings settings) {
-                    if (XContentMapValues.nodeBooleanValue(propNode)) {
-                        builder.enableGeoHash(true);
-                    }
-                    iterator.remove();
-                } else if (propName.equals("precision_step")) {
-                    builder.precisionStep(XContentMapValues.nodeIntegerValue(propNode));
-                    iterator.remove();
-                } else if (propName.equals("geohash_precision")) {
-                    if (propNode instanceof Integer) {
-                        builder.geoHashPrecision(XContentMapValues.nodeIntegerValue(propNode));
-                    } else {
-                        builder.geoHashPrecision(GeoUtils.geoHashLevelsForPrecision(propNode.toString()));
-                    }
-                    iterator.remove();
-                } else if (propName.equals(Names.IGNORE_MALFORMED)) {
-                    builder.ignoreMalformed(XContentMapValues.nodeBooleanValue(propNode));
-                    iterator.remove();
-                } else if (indexCreatedBeforeV2_0 && propName.equals("validate")) {
-                    builder.ignoreMalformed(!XContentMapValues.nodeBooleanValue(propNode));
-                    iterator.remove();
-                } else if (indexCreatedBeforeV2_0 && propName.equals("validate_lon")) {
-                    builder.ignoreMalformed(!XContentMapValues.nodeBooleanValue(propNode));
-                    iterator.remove();
-                } else if (indexCreatedBeforeV2_0 && propName.equals("validate_lat")) {
-                    builder.ignoreMalformed(!XContentMapValues.nodeBooleanValue(propNode));
-                    iterator.remove();
-                } else if (propName.equals(Names.COERCE)) {
-                    builder.coerce(XContentMapValues.nodeBooleanValue(propNode));
-                    iterator.remove();
-                } else if (indexCreatedBeforeV2_0 && propName.equals("normalize")) {
-                    builder.coerce(XContentMapValues.nodeBooleanValue(propNode));
-                    iterator.remove();
-                } else if (indexCreatedBeforeV2_0 && propName.equals("normalize_lat")) {
-                    builder.coerce(XContentMapValues.nodeBooleanValue(propNode));
-                    iterator.remove();
-                } else if (indexCreatedBeforeV2_0 && propName.equals("normalize_lon")) {
-                    builder.coerce(XContentMapValues.nodeBooleanValue(propNode));
-                    iterator.remove();
-                } else if (parseMultiField(builder, name, parserContext, propName, propNode)) {
-                    iterator.remove();
-                }
-            }
-            return builder;
+            return super.parse(name, node, parserContext);
         }
     }
 
-    public static final class GeoPointFieldType extends MappedFieldType {
+    public static final class GeoPointFieldType extends BaseGeoPointFieldType {
```

Comment:
What is this subclass doing, other than overriding the value function? Why does the behavior of value need to change from the base (ie why do we need this field type at all)?